src/System.ServiceModel.Primitives/tests/ServiceModel/SynchronizedCollection.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+using Infrastructure.Common;
+using Xunit;
+
+public class SynchronizedCollectiontest
+{
+#if FULLXUNIT_NOTSUPPORTED
+    [Fact]
+#endif
+    [WcfFact]
+    // This Unit test is ported from full framework test
+    public static void SynchronizedCollectionPublicMembersTest()
+    {
+        SynchronizedCollection<int> coll = new SynchronizedCollection<int>();
+        int size = 100;
+        for (int i = 0; i < size; i++)
+            coll.Add(i);
+
+        Assert.True(coll.Count == size, "collection count was wrong! got: " + coll.Count);
+
+        for (int i = 0; i < size; i++)
+        {
+            Assert.True(coll[i] == i, "coll element " + i + " was wrong!  got: " + coll[i]);
+            Assert.True(coll.IndexOf(i) == i, "coll IndexOf wasn't right!  i: " + i + "  got: " + coll.IndexOf(i));
+            Assert.True(coll.Contains(i), "coll Contains failed to find a value.");
```

Comment:
It would be good if each of these Assert strings said more about the value expected or found.  If these failed in the lab we wouldn't know why.